src/main/java/org/spongepowered/common/data/provider/entity/ServerPlayerData.java

```diff
@@ -76,7 +91,12 @@ public static void register(final DataProviderRegistrator registrator) {
                             h.bridge$setHealthScale(v);
                             return true;
                         })
-                        .resetOnDelete(Constants.Entity.Player.DEFAULT_HEALTH_SCALE);
+                        .resetOnDelete(Constants.Entity.Player.DEFAULT_HEALTH_SCALE)
+                    .create(Keys.VIEW_DISTANCE)
+                        .get(ServerPlayerEntityBridge::bridge$getViewDistance)
+                    .create(Keys.SKIN_PARTS)
+                        .get(ServerPlayerEntityBridge::bridge$getSkinParts);
+                        // .set(ServerPlayerEntityBridge::bridge$setSkinParts);
```

Comment:
?

Reply:
I did a basic implementation for this one, so it theoretically works since it is just a data tracker field, but because the client can update the skin parts option at any time the behaviour is unpredictable so I didn't actually activate the set operation.

Reply:
Probably should do it as read-only then.